extensions/wikia/AdEngine/js/ml/model/decisionTreeClassifier.js

```diff
@@ -0,0 +1,70 @@
+/*global define*/
+define('ext.wikia.adEngine.ml.model.decisionTreeClassifier', [
+	'wikia.log'
+], function () {
+	'use strict';
+
+	function buildUrl(modelId) {
+		return '/wikia.php?controller=AdEngine2Api&method=getModelData&id=' + modelId;
+	}
+
+	function findMax(nums) {
+		var index = 0;
+		for (var i = 0; i < nums.length; i++) {
+			index = nums[i] > nums[index] ? i : index;
+		}
+
+		return index;
+	}
+
+	function create(modelId) {
+		var lChilds, rChilds, thresholds, indices, classes;
+
+		function loadParameters(callback) {
+			var request = new XMLHttpRequest();
+
+			request.onreadystatechange = function() {
+				if (this.readyState === 4 && this.status === 200) {
+					var params = JSON.parse(this.responseText);
+
+					lChilds = params.lChilds;
+					rChilds = params.rChilds;
+					thresholds = params.thresholds;
+					indices = params.indices;
+					classes = params.classes;
+
+					callback(params);
+				}
+			};
+
+			request.open('GET', buildUrl(modelId), true);
+			request.send();
```

Comment:
Did you consider using `$.nirvana.sendRequest()`? I have a feeling that we're reinventing a wheel here :)

Reply:
There is no $.nirvana on mobile-wiki :(